### PR TITLE
Refactor path's module

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: fs/path/mod.ts [hunk 1]
```diff
@@ -271,7 +268,7 @@ export const win32 = {
     );
   },
 
-  normalize: function normalize(path: string) {
+  normalize(path: string) {
```
Comment (ry) [inline fs/path/mod.ts]: Add return type

File: fs/path/mod.ts [hunk 2]
```diff
@@ -384,7 +381,7 @@ export const win32 = {
     }
   },
 
-  isAbsolute: function isAbsolute(path: string) {
+  isAbsolute(path: string) {
```
Comment (ry) [inline fs/path/mod.ts]: Add return type. Here and elsewhere.

Comment (MaxGraey) [reply fs/path/mod.ts]: Done! Usually typescript greatly infer return type but I agree better define this explicitly.

File: fs/path/mod.ts [hunk 3]
```diff
@@ -1068,27 +1058,27 @@ export const posix = {
...
-  join: function join(...paths: string[]) {
-    if (arguments.length === 0) return ".";
-    let joined: string;
-    for (let i = 0; i < arguments.length; ++i) {
-      let arg = arguments[i];
-      assertPath(arg);
-      if (arg.length > 0) {
-        if (joined === undefined) joined = arg;
-        else joined += `/${arg}`;
+  join(...paths: string[]) {
```
Comment (ry) [inline fs/path/mod.ts]: return type here.

ideally we should have a lint that checks this.

**Pull request merged**